app/models/user.rb

```diff
@@ -10,9 +12,13 @@ class User < ApplicationRecord
 
   has_many :posts, dependent: :destroy
 
-  has_many :friend_requests_as_requestor, foreign_key: :requestor_id, class_name: :FriendRequest
-  has_many :friend_requests_as_receiver, foreign_key: :receiver_id, class_name: :FriendRequest
+  has_many :friend_requests, dependent: :destroy
+  has_many :pending_friends, through: :friend_requests, source: :friend
 
-  has_many :friendships, ->(user) { where("friend_a_id = ? OR friend_b_id = ?", user.id, user.id) }
+  has_many :friendships, dependent: :destroy
   has_many :friends, through: :friendships
+
+  def remove_friend(friend)
+    current_user.friends.destroy(friend)
+  end
```

Comment:
How come we have access to `current_user` in Model?

It is only available in Controllers or Views